pkg/serverless/serverless.go

```diff
@@ -138,6 +139,18 @@ func WaitForNextInvocation(stopCh chan struct{}, daemon *daemon.Daemon, id regis
 			metricTags = tags.AddInitTypeTag(metricTags)
 			metrics.SendTimeoutEnhancedMetric(metricTags, daemon.MetricAgent.Demux)
 			metrics.SendErrorsEnhancedMetric(metricTags, time.Now(), daemon.MetricAgent.Demux)
+
+			if ok := daemon.IsExecutionSpanComplete(); !ok {
+				ecs := daemon.ExecutionContext.GetCurrentState()
+				log.Debug("No hit on serverless.EndInvocation route. Attempting to finish Lambda execution span.")
+				timeoutContext := &invocationlifecycle.TimeoutExecutionInfo{
+					RequestID:       ecs.LastRequestID,
+					Runtime:         ecs.Runtime,
+					IsColdStart:     coldStartTags.IsColdStart,
+					IsProactiveInit: coldStartTags.IsProactiveInit,
+				}
+				daemon.InvocationProcessor.OnTimeoutInvokeEnd(timeoutContext)
```

Comment:
I really think we should explore using the already existing `OnInvokeEnd` method since this shares a ton of code.